project/HaHaFi/proposal_2/README.md

```diff
@@ -0,0 +1,64 @@
+# Auto video caption and download 
+## Introdution
+This is a tool which want to download the video on the web page, and get the English caption of the video without using any speech-to-text api. Just use your local GPU.
+
+The tool has two part.
+1. Get the m3u8 url in web page and download the video (in C++).
+2. Use speech recognition model to get the caption of the video file (in Python).
+
+**!!Note!!**
+*  The speech recognition is only for English.
+
+
+## Problem to solve
+1. Download the m3u8 video with specify protocol.
+2. The performance of speech recognition model is important. 
```

Comment:
Do you plan to implement the recognition model yourself?  If not, the project has little to do with numerical calculation.

Reply:
In my original plan, I want to search some paper for using the state of the art recognition model, and use pytorch deeplearning framework. 
I think you are right. There is few numerical calculation work will be implemented by me in the plan.

Reply:
Using a framework will not provide the opportunity to implement numerical code.  Also, implementing from papers takes a lot of time even if you use an existing framework.  Neither improves your coding skills that the term project tries to help you develop.

Reply:
I see. I will discard this proposal, and make the another one complete.